cmd/cosign/cli/sign/sign_blob.go

```diff
@@ -117,26 +124,83 @@ func SignBlobCmd(ro *options.RootOptions, ko options.KeyOpts, payloadPath string
 		}
 		ui.Infof(ctx, "tlog entry created with index: %d", *entry.LogIndex)
 		signedPayload.Bundle = cbundle.EntryToBundle(entry)
+		rekorEntry = entry
 	}
 
 	// if bundle is specified, just do that and ignore the rest
 	if ko.BundlePath != "" {
-		signedPayload.Base64Signature = base64.StdEncoding.EncodeToString(sig)
+		if ko.UsePBBundleFormat {
+			vm := pbbundle.VerificationMaterial{}
 
-		certBytes, err := extractCertificate(ctx, sv)
-		if err != nil {
-			return nil, err
-		}
-		signedPayload.Cert = base64.StdEncoding.EncodeToString(certBytes)
+			certBytes, err := sv.Bytes(ctx)
```

Comment:
We'll also need to handle where `sv.Bytes()` returns a key, since that goes into a different message. We could change `sv.Bytes()` to return whether it's a key or cert, or try to parse the return value to determine if it's a key or cert.